### PR TITLE
fixed scrolling of pages when navbar is open and close navbar from outside in desktop

### LINKED ISSUE #1190: Navbar overlapping with the content on state pages
**Describe the bug**
With the navbar open if I go to any state's page through the search bar, the navbar should close automatically but it's not happening right now on the desktop instead it's blocking the content.

**To Reproduce**

1. Open Navbar by clicking on Menu and keep it open.
2. Search for any city or state and go to the states page you will see that navbar is blocking the content.

**Expected behavior**
When I am going to state's page with the navbar is open it should close automatically.

**Screenshots**
Screenshot

**Desktop (please complete the following information):**

- OS: Ubuntu 18.04 and Windows 10
- Browser: Chrome
- Version: 81.0.4044.92 (Official Build) (64-bit)

### LINKED ISSUE #1311: menu sidebar scroll on mobile screen
**Describe the bug**
the menu bar in mobile phone when you click the sidebar is scrollable but it doesnt cover the whole screen its half way so it creates a problem, like i am seeing half menu info and half the back page info.
even if you scroll down you wont be able to close the sidebar as the close button goes up when you scroll.
**To Reproduce**

1. Go to 'mobile'
2. Click on 'menu'
3. Scroll down to the options then you will see half the sidebar of menu ends and main page starts from half
image

**Expected behavior**
so the behavior should have been the side menu bar when you scroll also it should only show the menu options thing  it shoul not show the main page after half scroll.
or else when you half scroll and the scrolling sidebar menu options are over if you click below to that it should close automatically the menu sidebar.
so we can keep the close fixed at top when you make the sidebar full screen or after the sidebar area is done if you click anywhere automatically it closes.
**Screenshots**
image

**Smartphone (please complete the following information):**

- Device: [any device]
- Browser [any]

**Additional context**

### PR DESCRIPTION
**Pull request by Binu42**
**Description of PR**
* fixed scrolling of pages when navbar is opened.
* close navbar from outside click of expand in desktop.
* close navbar on click of covid19India in navbar otherwise it going to homepage and still navbar is open.
* show active page in navbar.

**Type of PR**

- [x] Bugfix

**Relevant Issues**  
Fixes #1190, Fixes #1311 

**Checklist**

- [x] Compiles and passes lint tests
- [x] Properly formatted
- [x] Tested on desktop
- [x] Tested on phone

**Demo**
#### Before Fix (Desktop)
desktop demo before
<br>

#### After Fix (Desktop)
desktop demo after
<br>

#### Before Fix (Mobile)
mobile demo before
<br>

#### After Fix (Mobile)
mobile demo after

### CODE REVIEW

File: src/components/navbar.js [hunk 1]
```diff
@@ -11,6 +11,11 @@ const navLinkProps = (path, animationDelay) => ({
 function Navbar({pages}) {
   const [expand, setExpand] = useState(false);
 
+  useEffect(() => {
+    if (expand) document.body.classList.add('modal-open');
```
Comment (jeremyphilemon) [inline src/components/navbar.js]: Hey thanks! Can you make the classes self-documenting as well, like "scroll-disabled" or something like that, which would also support the current modals?

**Review by jeremyphilemon — CHANGES_REQUESTED** [review]
Hey it looks like this has caused misalignment in the navbar on mobiles, and I'm still able to scroll down from the expanded navbar? (I tried this on my laptop with responsive mode on maybe I should try this on the phone to work?)

Screen Shot 2020-04-17 at 12 09 40 PM

Otherwise we're good, can you look into what may be the problem?

File: src/components/navbar.js (hunk 1 above)
Comment (Binu42) [reply src/components/navbar.js]: okay for sure...

**Comment by Binu42** [comment]
> Hey it looks like this has caused misalignment in the navbar on mobiles, and I'm still able to scroll down from the expanded navbar? (I tried this on my laptop with responsive mode on maybe I should try this on the phone to work?)
> 
> Screen Shot 2020-04-17 at 12 09 40 PM
> 
> Otherwise we're good, can you look into what may be the problem?

@jeremyphilemon thanks for your review i have changed the things accordingly. and i'm not able to see any misalignment in navbar on mobiles and i'm not able to scroll when navbar is open please check it once the recent link for now is https://covid19india-react-df2amcylo.now.sh and let me know if any issues.
i have also added closing of expanded navbar when user click on covid19India (at navbar top) otherwise currently in website it's going to homepage but not closing navbar. i hope that will be fine.

**Comment by Binu42** [comment]
@jeremyphilemon i noticed misalignment in navbar little late it's fixed now, thanks.

**Review by jeremyphilemon — CHANGES_REQUESTED** [review]
Looks good.

Some issues I still found:
- The home screen seems to be jumping when the navbar is toggled on desktop
- Let's not have the backdrop bg as blue
- If you're highlighting the options like this:
Screen Shot 2020-04-18 at 2 17 14 AM
  it'd be better if you could also have the font change it's color to $pblue.


Also squash your commits and rebase, please :)

**Comment by Binu42** [comment]
@jeremyphilemon i have changed backdrop bg to transparent dark and active page navbar color to $pblue also taken care of jumping on pages when navbar toggles. have a look on recent deployment.

**Comment by jeremyphilemon** [comment]
@Binu42 Hey thanks for the quick turnaround!

It still seems to jump around for me on safari (macOS) :/ Could you try preserving this design instead, it doesn't have to be filled even of the left – it was left like that intentionally. 

Screen Shot 2020-04-18 at 9 55 50 PM

Thanks :)

**Comment by Binu42** [comment]
@jeremyphilemon i've removed my last commits related with nav-right btn (menu & close).

**Comment by Binu42** [comment]
@jeremyphilemon can u look for this also and give some feedback.

**Comment by jeremyphilemon** [comment]
Patched by #1364

**Pull request closed without merging**